web3/providers/persistent/request_processor.py

```diff
@@ -217,6 +232,28 @@ def get_request_information_for_response(
 
         return request_info
 
+    def append_result_formatter_for_request(
+        self, request_id: int, result_formatter: Callable[..., Any]
+    ) -> None:
+        cache_key = generate_cache_key(request_id)
+        cached_request_info_for_id: RequestInformation = (
+            self._request_information_cache.get_cache_entry(cache_key)
+        )
+        if cached_request_info_for_id is not None:
+            cached_request_info_for_id.response_formatters = (
+                compose(
+                    result_formatter,
+                    cached_request_info_for_id.response_formatters[0],
```

Comment:
What do you think about making these `response_formatters` their own attributes instead of a tuple? Then we could reference like:
```suggestion
                    cached_request_info_for_id.result_formatters,
                    cached_request_info_for_id.error_formatters,
                    ...
```

Reply:
Ah, yeah that could make this more readable for sure. I like this. I'll play around with that idea.

Reply:
Since I think it's only in this part of the code base, what do you think about [unpacking the values](https://github.com/ethereum/web3.py/pull/3370/commits/56adab130945a29fd5f4d62990f54557c4792889) for better readability?

Reply:
That works for me!